src/include/storage/index/hash_index.h

```diff
@@ -25,6 +32,7 @@ class OnDiskHashIndex {
     virtual void prepareRollback() = 0;
     virtual void checkpointInMemory() = 0;
     virtual void rollbackInMemory() = 0;
+    virtual void bulkReserve(uint64_t numValuesToAppend) = 0;
```

Comment:
Yes, it's being used so that the HashIndex<T> only needs a type when it is accessed, and can be stored as a `unique_ptr<OnDiskHashIndex>` (and it simplifies a few functions that don't need type information).